File: rb/features.rb
# ThurgauPopulationData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ThurgauPopulationDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      ThurgauPopulationDataBaseFeature.new
    when "test"
      ThurgauPopulationDataTestFeature.new
    else
      ThurgauPopulationDataBaseFeature.new
    end
  end
end

File: py/features.py
# ThurgauPopulationData SDK feature factory

from feature.base_feature import ThurgauPopulationDataBaseFeature
from feature.test_feature import ThurgauPopulationDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ThurgauPopulationDataBaseFeature(),
        "test": lambda: ThurgauPopulationDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
